spec/api-web-contents-spec.js

```diff
@@ -708,4 +708,50 @@ describe('webContents module', () => {
       w.loadURL(`file://${fixtures}/pages/a.html`)
     })
   })
+
+  describe('referrer', () => {
+    it('propagates referrer information to new target=_blank windows', (done) => {
+      const server = http.createServer((req, res) => {
+        if (req.url === '/should_have_referrer') {
+          assert.equal(req.headers.referer, 'http://127.0.0.1:' + server.address().port + '/')
+          return done()
+        }
+        res.end('<a id="a" href="/should_have_referrer" target="_blank">link</a>')
+      })
+      server.listen(0, '127.0.0.1', () => {
+        const url = 'http://127.0.0.1:' + server.address().port + '/'
+        w.webContents.once('did-finish-load', () => {
+          w.webContents.once('new-window', (event, newUrl, frameName, disposition, options, features, referrer) => {
+            assert.equal(referrer.url, url)
+            assert.equal(referrer.policy, 'no-referrer-when-downgrade')
+          })
+          w.webContents.executeJavaScript('a.click()')
+        })
+        w.loadURL(url)
+      })
+    })
+
+    // TODO(jeremy): window.open() in a real browser passes the referrer, but
+    // our hacked-up window.open() shim doesn't. It should.
+    xit('propagates referrer information to windows opened with window.open', (done) => {
```

Comment:
Is the plan to work on fixing this spec in this PR or separate ? Rest of the code LGTM.

Reply:
I think it would make sense to land this and fix the `window.open` issue separately, since it'll be almost entirely unrelated code and this PR at least improves the situation somewhat. Should I remove the `xit` test?

Reply:
I think its fine to have the test.